### ISSUE TITLE
Add mouse wheel scroll block while hovering canvas of HTML5 build

### ISSUE THREAD

**Issue by subsoap** [defold/defold#5636]
Adding this to the bottom of scripts will prevent mouse wheel from scrolling a browser window while hovering cursor on the canvas. Important for games which use mouse wheel for whatever reason.

```
document.getElementById("canvas").onwheel = function(event) {
    event.preventDefault();
};

document.getElementById("canvas").onmousewheel = function(event) {
    event.preventDefault();
};
```

This has this code included in its template

https://subsoap.itch.io/defold-shaders-examples

This does not

https://subsoap.itch.io/shadertests

**Comment by britzl** [comment]
Yes, that's a great addition! Why not [create a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork)?

The file in question is this one: https://github.com/defold/defold/blob/dev/engine/engine/content/builtins/manifests/web/engine_template.html

**Comment by AGulev** [comment]
but I'm sure Defold has this logic already   https://github.com/defold/defold/blob/cc9389b14e0ebd9edea8c3f37e72e35b42808f97/engine/glfw/js/library_glfw.js#L409

Is there any troubles with that?

**Comment by britzl** [comment]
It's definitely a difference in behaviour between the two links Pkeod provided.

**Comment by AGulev** [comment]
Ah, ok. I see it now (in chrome, safari is fine)

I think it's because we use this gate:

>if (!GLFW.isCanvasActive(event)) { return; }

so, we don't take into account hovering, only when canvas is active.
Also, we don't use `onwheel` only `onmousewheel `

**Comment by subsoap** [comment]
@AGulev There will be design cases where people want to be able to allow mouse scroll when hovering the Defold engine canvas. But seems like there are already browser inconsistencies in that behavior? I can see Defold eventually being used for non-game design elements on website and people would not want the mouse wheel behavior to be always blocked on focus / hover.

**Comment by britzl** [comment]
Hmm, it is a bit problematic that some of the behavior is in the engine_template.html and some in library_glfw.js. I don't like that at all. It should preferably be in one place or the other.

And I don't really like that we have a lot of resize logic in the engine_template.html either. I would prefer if the requirement on the .html file would be as minimal as possible. Preferably only a `<canvas>` tag, one `<script src="dmloader.js">` tag and a single javascript function call.

**Comment by subsoap** [comment]
Internalizing most of these core features while cleaning up the .html, having game.project config toggles for what users may want to toggle would be best.
